session/session.go

```diff
@@ -2610,7 +2623,7 @@ func (s *session) loadCommonGlobalVariablesIfNeeded() error {
 		// `collation_server` is related to `character_set_server`, set `character_set_server` will also set `collation_server`.
 		// We have to make sure we set the `collation_server` with right value.
 		if _, ok := vars.GetSystemVar(varName); !ok || varName == variable.CollationServer {
-			err = variable.SetSessionSystemVar(s.sessionVars, varName, varVal)
+			err = vars.SetSystemVar(varName, varVal)
```

Comment:
Not for reviewers - this is a slight behavior change:
`SetSessionSystemVar` applies validation, and then sets the variable. `vars.SetSystemVar` just sets the variable.

This is required because the validate function will validate scope, and refuse some of the items because `loadCommonGlobalVariablesIfNeeded` contains some global variables, for which the `SetSession` functions as an Init function.